### DISCUSSION TITLE
max()如何判斷list中的字串大小

### DISCUSSION THREAD

**Discussion by ghost** [1101-GS4719-C/Python-Community-GS4719-C#854] [Ideas]
如題，我發現他在判斷大小時不是以字串長度，而是以字母排序，如圖所示，list1中的字串字首b的順序靠後，所以結果為b；至於其他，像數字順序比英文還前；中文的順序比英文還後面等。複習講義發現的發現的有趣小問題，幫大家複習。

![螢幕擷取畫面 2022-11-22 211854](https://user-images.githubusercontent.com/113499571/203323867-4d8a54d7-8791-459a-b57e-cd0accc2b8dd.png)

**Comment by dabbie109303501** [comment]
雖然我不是很確定但它的排序應該是依照ascii碼排的
![image](https://user-images.githubusercontent.com/75252877/204443891-92110d83-3ad5-4f33-b28f-74c774806850.png)
然後因為中文不在裡面 所以放最後

**Comment by williamyen0623** [comment]
使用的時候還真的會常常忘記因而避開使用或是又要重新複習，邏輯大概就像樓上所說的ASCII碼排序吧

**Comment by ching020721** [comment]
謝謝分享，如果忘記的話可能就是再查資料，多使用最後總會記住的吧XD

**Comment by manipulateed** [comment]
原來list 能透過這方式去比字串的大小，雖然用到的機會不多，但能增加知識也不錯，

**Comment by evanchou1204** [comment]
謝謝分享，很多程式的東西真的太久沒用就容易忘記，畢竟人不是電腦，不可能把全部的指令都背起來。

**Comment by StevenWanglolz** [comment]
感謝分享! 這種細節各種程式語言都有好多，很容易忘記!

**Comment by CPCpyt** [comment]
我剛好也有這樣的問題，感謝分享及討論！

**Comment by RogelioKG** [comment]
可以理解成「按 ASCII 排序」。
不過嚴謹一點說，如果是 Python 3，
因為包含中文，應該說「按 Unicode 碼位排序」比較妥當。
